sys/drv/stdvga.c

```diff
@@ -173,7 +173,7 @@ static int stdvga_attach(device_t *dev) {
 
   stdvga->mem =
     bus_alloc_resource_any(dev, RT_MEMORY, 0, RF_ACTIVE | RF_PREFETCHABLE);
-  stdvga->io = bus_alloc_resource_any(dev, RT_MEMORY, 2, RF_ACTIVE);
```

Comment:
Why we need to change `rid` here?

Reply:
Now all valid bars of a device appear in sequence. 
Before this fix, the VGA driver had to be aware of the fact that we sense the VGA bars as 
valid, invalid, valid, invalid, invalid, and invalid.
Since VGA only uses two bars, after the fix, bar 2 (and the rest) are invalid and therefore cannot be used.

Reply:
Thanks for your explanation.